effect/shared/src/main/scala/com/monovore/decline/effect/CommandIOApp.scala

```diff
@@ -0,0 +1,42 @@
+package com.monovore.decline.effect
+
+import cats.effect.{IO, IOApp, ExitCode}
+import cats.implicits._
+
+import com.monovore.decline._
+
+abstract class CommandIOApp(name: String, header: String, helpFlag: Boolean = true, version: String = "") extends IOApp {
+
+  def commandOpts: Opts[IO[ExitCode]]
+
+  private[this] val command: Command[IO[ExitCode]] = {
```

Comment:
Now that I look at this with fresh eyes, this `val` is a code smell since it is depending on a `def` and could cause initialization order errors. It should be either a `lazy val` or a `def`, the former giving just the benefit of being synchronized... which I doubt is of any gain in here... any strong thoughts on that?

Reply:
Slight preference for `def`.